lib/pcb/svg-object-fns/create-svg-objects-from-pcb-rat-nests.ts

```diff
@@ -0,0 +1,149 @@
+import { getFullConnectivityMapFromCircuitJson } from "circuit-json-to-connectivity-map"
+import type { AnyCircuitElement } from "circuit-json"
+import { type INode as SvgObject } from "svgson"
+import { type Matrix, applyToPoint } from "transformation-matrix"
+
+interface RatsNestLine {
+  key: string
+  startPoint: { x: number; y: number }
+  endPoint: { x: number; y: number }
+  isInNet: boolean
+}
+
+export function createSvgObjectsForRatsNest(
+  soup: AnyCircuitElement[],
+  transform: Matrix,
+): SvgObject[] {
+  // Build a lookup map by cleaned ID for any element that might provide a position.
+  const elementMap = new Map<string, AnyCircuitElement>()
+  for (const elm of soup) {
+    const id =
+      (elm as any).id || (elm as any).pcb_port_id || (elm as any).source_port_id
+    if (id) {
+      const cleanedId = id.replace(/_\d+$/, "")
+      elementMap.set(cleanedId, elm)
```

Comment:
the ids aren't always going to be in the format `source_port_...`, this code will fail in weird ways (in general, it's best to avoid regex'ing or string manipulation)

I think you can look at the connectivity map to find all the connected ids then iterate over them instead?